client/src/main/java/org/apache/nemo/client/beam/NemoPipelineResult.java

```diff
@@ -49,10 +63,18 @@ public State cancel() throws IOException {
 
   @Override
   public State waitUntilFinish(final Duration duration) {
-    throw new UnsupportedOperationException();
-    // TODO #208: NemoPipelineResult#waitUntilFinish hangs
-    // Previous code that hangs the job:
-    // return (State) super.waitUntilJobFinish(duration.getMillis(), TimeUnit.MILLISECONDS);
+    final boolean finished;
+    try {
+      finished = this.jobDone.await(duration.getMillis(), TimeUnit.MILLISECONDS);
```

Comment:
May I please ask why we are using jobDone await instead of thread sleep?

Reply:
There's no particular reason. If you have any suggestions, you may suggest changes.